src/data/move-attrs/chance-based-move-effect-attr.ts

```diff
@@ -53,23 +53,60 @@ export abstract class ChanceBasedMoveEffectAttr extends MoveEffectAttr {
    * @returns The final percent chance of this attribute's effect applying. If negative, the
    * effect is guaranteed to apply.
    */
-  public getMoveChance(
-    user: Pokemon,
-    target: Pokemon,
-    move: Move,
-    selfEffect: boolean,
-    showAbility: boolean = false,
-  ): number {
+  public getMoveChance(user: Pokemon, target: Pokemon, move: Move, showAbility: boolean = false): number {
     const moveChance = new NumberHolder(this.effectChanceOverride ?? move.chance);
 
     applyAbAttrs(MoveEffectChanceMultiplierAbAttr, user, false, moveChance, move, showAbility);
 
     const userSide = user.getArenaTagSide();
     globalScene.arena.applyTagsForSide(ArenaTagType.WATER_FIRE_PLEDGE, userSide, false, moveChance);
 
-    if (!selfEffect) {
+    if (!this.selfTarget) {
       applyAbAttrs(IgnoreMoveEffectsAbAttr, target, false, user, move, moveChance);
     }
     return moveChance.value;
   }
+
+  /**
+   * Calculates effect score, factoring in this attribute's {@linkcode getRawEffectScore | raw effect score}
+   * and {@linkcode getMoveChance | chance to apply}.
+   */
+  public override getEffectScore(user: EnemyPokemon, target: Pokemon, move: Move): number {
+    /**
+     * The attribute's chance to apply its effect
+     * @todo this chance calculation may prematurely reveal abilities
+     */
+    const chance = this.getMoveChance(user, target, move);
+    /** The attribute's effect score, assuming its effect always applies */
+    const rawScore = this.getRawEffectScore(user, target, move);
+
+    /**
+     * The attribute's effect score after factoring in effect chance.
+     * This may be a decimal number; the final output is either
+     * `floor(chanceWeightedScore)` or `floor(chanceWeightedScore) + 1`
+     */
+    const chanceWeightedScore = chance * rawScore;
+    /** The minimum integer score this function can return */
+    const minScore = Math.floor(chanceWeightedScore);
+    /**
+     * The chance to return `minScore + 1` instead of `minScore`.
+     * This is the decimal component of `chanceWeightedScore` scaled
+     * up to a percent value, then rounded down.
+     */
+    const tierUpChance = Math.floor((chanceWeightedScore % 1) * 100);
+
+    return this.getRandomScore(user, tierUpChance, minScore + 1, minScore);
```

Comment:
```suggestion
    return minscore + this.getRandomScore(user, tierUpChance);
```
Since `getRandomScore` was changed.